### PR TITLE
Update SDK for new API version

### PR DESCRIPTION
**Pull request by emmeliaAra**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: common/ManagementTestShared/Redesign/ManagementRecordedTestBase.cs [hunk 1]
```diff
@@ -169,7 +169,8 @@ private ArmEnvironment GetArmEnvironment(TestEnvironment environment)
             if (baseUri == ArmEnvironment.AzureGovernment.Endpoint)
                 return ArmEnvironment.AzureGovernment;
 
-            return new ArmEnvironment(new Uri(endpoint), SessionEnvironment.ServiceManagementUrl ?? $"{endpoint}/.default");
+            //return new ArmEnvironment(new Uri(endpoint), SessionEnvironment.ServiceManagementUrl ?? $"{endpoint}/.default");
+            return new ArmEnvironment(new Uri(endpoint), environment.ServiceManagementUrl ?? $"{endpoint}/.default");
```
Comment (archerzz) [inline common/ManagementTestShared/Redesign/ManagementRecordedTestBase.cs:173]: This is used for all mgmt sdk tests. Please roll back the change, and use other ways to get `ServiceManagementUrl`.

File: sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/CHANGELOG.md [hunk 2]
```diff
@@ -10,7 +10,7 @@
 
 ### Other Changes
 
-## 1.0.0-beta.1 (2023-10-30)
+## 1.0.0-beta.1 (2023-12-30)
```
Comment (archerzz) [inline sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/CHANGELOG.md:13]: `1.0.0-beta.1` has already been published: https://www.nuget.org/packages/Azure.ResourceManager.HardwareSecurityModules/1.0.0-beta.1

You need to publish `1.0.0-beta.2`. The entry is already added in the `CHANGELOG.md`. And don't forget to update the verison in `.csproj` file.

**Review by archerzz — CHANGES_REQUESTED** [review]

File: common/ManagementTestShared/Redesign/ManagementRecordedTestBase.cs (hunk 1 above)
Comment (emmeliaAra) [reply common/ManagementTestShared/Redesign/ManagementRecordedTestBase.cs:173]: I've used this to be able to run the test in DF environment and I pushed this change accidentally. rolled back, thanks.

File: sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/CHANGELOG.md (hunk 2 above)
Comment (emmeliaAra) [reply sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/CHANGELOG.md]: @archerzz - Updated the readme file, thank you. The .csproj file is already using  <Version>1.0.0-beta.2</Version>, so I did not update this

**Comment by emmeliaAra** [comment]
Please **DO NOT MERGE!** Swagger for this version is reviewed but not merged yet, since feature is still under development! https://github.com/Azure/azure-rest-api-specs/pull/26326

File: sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/autorest.md [hunk 3]
```diff
@@ -17,6 +17,7 @@ sample-gen:
 skip-csproj: true
 modelerfour:
   flatten-payloads: false
+  lenient-model-deduplication: true
```
Comment (archerzz) [inline sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/autorest.md:20]: This flag is deprecated. Do you still want to use it? https://github.com/Azure/autorest/blob/fe71a68f5f65dcb48340a6bb350e9f583b838571/packages/extensions/modelerfour/readme.md?plain=1#L49-L58

**Review by archerzz — COMMENTED** [review]
@emmeliaAra Generally it's fine. Just want to double check if you want to use a deprecated flag `lenient-model-deduplication`.

File: sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/autorest.md (hunk 3 above)
Comment (emmeliaAra) [reply sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/autorest.md:20]: Without the modelerfour flag we are getting the error below. I've submitted a Swagger PR a while back to fix the duplicated Schema "Resource" but I am still getting this error when running dotnet build /t:GenerateCode command, thus I had to add the tag to be able to generate the code. Any suggestions to resolve this ? 

**PR with the fix for this issue:** https://github.com/Azure/azure-rest-api-specs/pull/24425

**Error when Generating Code without modelerfour tag:** 
image

Comment (archerzz) [reply sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/autorest.md:20]: @emmeliaAra There is a `DedicatedHsmResource` which looks like a `Resource` or `TrackedResource`: https://github.com/Azure/azure-rest-api-specs/blob/9a3161dbc683120d907689209a6eebd450af8c3d/specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/stable/2021-11-30/dedicatedhsm.json#L807

Also note that there is a [custom `SystemData`](https://github.com/Azure/azure-rest-api-specs/blob/9a3161dbc683120d907689209a6eebd450af8c3d/specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/stable/2021-11-30/dedicatedhsm.json#L858) which looks like the one in `common-types`: https://github.com/Azure/azure-rest-api-specs/blob/c1ba9df47907f9012ae14ca4616aed9e5665f9e5/specification/common-types/resource-management/v5/types.json#L502

Better to use the common types as many as possible.

Comment (emmeliaAra) [reply sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/autorest.md:20]: Hi, @archerzz - Thanks for pointing out the issue. The issue comes from the definition in DHSM swagger and the new version we are releasing is for Cloud HSM. We are planning to release a new version for both services when we move CHSM to stable folder and I will make sure to make this change in the next PR, if we can go ahead with this one as it is.

Comment (archerzz) [reply sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/autorest.md:20]: @emmeliaAra gotcha

**Review by archerzz — APPROVED** [review]

**Pull request merged**